sklearn/model_selection/tests/test_split.py

```diff
@@ -1582,3 +1504,74 @@ def test_leave_p_out_empty_trainset():
             ValueError,
             match='p=2 must be strictly less than the number of samples=2'):
         next(cv.split(X, y, groups=[1, 2]))
+
+
+@pytest.mark.parametrize('Klass', [
+    KFold,
+    StratifiedKFold,
+    RepeatedKFold,
+    RepeatedStratifiedKFold,
+    ShuffleSplit,
+    StratifiedShuffleSplit,
+    GroupShuffleSplit
+])
+@pytest.mark.parametrize('random_state', [0, None, np.random.RandomState(0)])
+def test_random_state_reproducibility(Klass, random_state):
+    # Randomized CV iterators should consistently yield the same splits for
+    # different calls to split(), no matter the random state.
+    X = np.arange(20).reshape(-1, 1)
+    y = [0] * 10 + [1] * 10
+    groups = np.arange(20)
+
+    kwargs = {}
+    if Klass in (KFold, StratifiedKFold):
+        # The other classes don't have a shuffle parameter
+        kwargs = {'shuffle': True}
+    cv = Klass(3, random_state=random_state, **kwargs)
+
+    expected = list(cv.split(X, y, groups))
+    for _ in range(10):
+        np.testing.assert_equal(list(cv.split(X, y, groups)), expected)
+
+
+@pytest.mark.parametrize('Klass', [
+    KFold,
+    StratifiedKFold,
+    RepeatedKFold,
+    RepeatedStratifiedKFold,
+    ShuffleSplit,
+    StratifiedShuffleSplit,
+    GroupShuffleSplit
+])
+@pytest.mark.parametrize('random_state', [0, None, np.random.RandomState(0)])
+def test_different_instances_splits(Klass, random_state):
+    # 2 different instances of a randomized splitter should yield the same
+    # splits if the same int is passed as random_state. If random_state is
+    # None or an instance, splits should be different.
```

Comment:
If its None or an instance, is there is small possibility that they will be the same? This would make this test flaky.

Reply:
yes but we only need 1 positive example to ensure the property.